unet-instance-segmentation/experiment/trainer.py

```diff
@@ -87,15 +101,92 @@ def __init__(self):
         self.min_contour_points = 20
 
     def prepare_data_for_training(self):
-        self._download_data()
+        self.segmentation_dataset = self.get_segmentation_dataset()
+        if self.segmentation_dataset is not None:
+            self.download_segmentation_data_into_masks()
+        else:
+            self.mask_dataset, self.image_dataset = self.get_mask_image_datasets()
+            if self.mask_dataset is not None and self.image_dataset is not None:
```

Comment:
They can't be None as it would raise an error in `get_mask_image_dataset`

Reply:
I think logging.error does not stop execution right? so I believe they can still be None in this case. 
I can add a raise exception to get_mask_dataset so that the exception halts the execution of the rest of the code in case the datasets are not found, then we can be sure they can't be None